web-registry/src/components/organisms/RolesForm/components/RoleField/RoleField.tsx

```diff
@@ -0,0 +1,188 @@
+import { forwardRef, useEffect, useState } from 'react';
+import {
+  ApolloClient,
+  NormalizedCacheObject,
+  useApolloClient,
+} from '@apollo/client';
+import { Autocomplete, TextField } from '@mui/material';
+import { useQuery } from '@tanstack/react-query';
+
+import FieldFormControl from 'web-components/lib/components/inputs/new/FieldFormControl/FieldFormControl';
+import { truncate } from 'web-components/lib/utils/truncate';
+
+import { getPartiesByAccountIdQuery } from 'lib/queries/react-query/registry-server/graphql/getPartiesByAccountIdById/getPartiesByAccountIdQuery';
+import { getPartiesByNameOrAddrQuery } from 'lib/queries/react-query/registry-server/graphql/getPartiesByNameOrAddr/getPartiesByNameOrAddrQuery';
+import { useWallet } from 'lib/wallet/wallet';
+
+import { ProfileModal } from '../ProfileModal/ProfileModal';
+import { ProfileModalSchemaType } from '../ProfileModal/ProfileModal.schema';
+import { useDebounce } from './hooks/useDebounce';
+import { useSaveProfile } from './hooks/useSaveProfile';
+import { AddNewProfile } from './RoleField.AddNewProfile';
+import { NoGroup } from './RoleField.NoGroup';
+import { ProfileGroup } from './RoleField.ProfileGroup';
+import { useStyles } from './RoleField.styles';
+import { getParties, group, isProfile } from './RoleField.utils';
+
+interface Props {
+  className?: string;
+  classes?: {
+    root?: string;
+  };
+  description?: string;
+  label?: string;
+  optional?: boolean | string;
+  setValue: (value: ProfileModalSchemaType | null) => void;
+  value?: ProfileModalSchemaType | null;
+}
+
+export const RoleField = forwardRef<HTMLInputElement, Props>(
+  (
+    {
+      className,
+      classes,
+      label,
+      optional,
+      description,
+      setValue,
+      value,
+    }: Props,
+    ref,
+  ) => {
+    const { classes: styles, cx } = useStyles();
+    const { accountId } = useWallet();
+    const [inputValue, setInputValue] = useState('');
+    const [options, setOptions] = useState<readonly ProfileModalSchemaType[]>(
+      [],
+    );
+    const [profileAdd, setProfileAdd] = useState<ProfileModalSchemaType | null>(
+      null,
+    );
+
+    const graphqlClient =
+      useApolloClient() as ApolloClient<NormalizedCacheObject>;
+    const debouncedInputValue = useDebounce(inputValue);
+    const { data: parties } = useQuery(
+      getPartiesByNameOrAddrQuery({
+        client: graphqlClient,
+        enabled: !!graphqlClient && !!debouncedInputValue,
+        input: debouncedInputValue,
+      }),
+    );
+    const { data: partiesByAccountId } = useQuery(
+      getPartiesByAccountIdQuery({
+        client: graphqlClient,
+        id: accountId,
+        enabled: !!accountId && !!graphqlClient,
+      }),
+    );
```

Comment:
I'm wondering if it's fine to have components doing fetch directly. While this is convenient it prevents us to add stories for them. This component would have been interesting to visualize and test with mock data in storybook (if it's too much of a pain to refactor we can leave it).

Reply:
we can extract `getPartiesByAccountIdQuery` easily but with regards to `getPartiesByNameOrAddrQuery`, it seems a bit more complex, since it depends on `deboucedInputValue`, hence `inputValue` which is part of the component state, unless we set it as a component prop instead? and have some in between component like this:

```
const [inputProjectDeveloper, setInputProjectDeveloper] = useState('');
const debouncedInputProjectDeveloper = useDebounce(inputProjectDeveloper);
const { data: partiesProjectDeveloper } = useQuery(
  getPartiesByNameOrAddrQuery({
    client: graphqlClient,
    enabled: !!graphqlClient && !!debouncedInputProjectDeveloper,
    input: debouncedInputProjectDeveloper,
  }),
);

return <RoleField
  setInputValue={setInputValue}
  inputValue={inputValue}
  parties={partiesProjectDeveloper}
  {...props}
/>
```

Reply:
ping @flagrede

Reply:
Sorry for the late answer, missed the notification.
I agree that the debounce value should be part of the component.
We could just pass the `setInputValue ` in your example and keep the debounce value in `RoleField`.
We would trigger the `setInputValue ` call in a `useEffect` in `RoleField` whenever the debounce value change.

Reply:
Oh, I see the confusion, I meant that `inputValue` stays in `RoleField` too and the callback that would be passed to `RoleField` would be from another `useState` from the component using `RoleField` dedicated to `getPartiesByNameOrAddrQuery` query. Does that make sense ?
The Flow would be:
inputValue change -> debounce value change -> call callback from parent component with the latest debounce value

Reply:
This is just to save one prop from `RoleField` and separate the component state from the query param but your suggestion works too.

Reply:
oh I see I think that works

Reply:
done, need to add a story for RoleField now but ready for another round of review
need the registry-server PR to be merged still before everything works as expected